pennylane/interfaces/torch.py

```diff
@@ -144,29 +201,18 @@ def backward(ctx, grad_output):  # pragma: no cover
 
             # evaluate the Jacobian matrix of the QNode
             jacobian = qnode.jacobian(ctx.args, ctx.kwargs)
+            jacobian = torch.as_tensor(jacobian, dtype=grad_output.dtype)
 
-            if grad_output.is_cuda:  # pragma: no cover
-                grad_output_np = grad_output.cpu().detach().numpy()
-            else:
-                grad_output_np = grad_output.detach().numpy()
-
-            # perform the vector-Jacobian product
-            if not grad_output_np.shape:
-                temp = grad_output_np * jacobian
-            else:
-                temp = grad_output_np.T @ jacobian
+            vjp = torch.transpose(grad_output.view(-1, 1), 0, 1) @ jacobian
```

Comment:
Note: this is actually a lot cleaner and better than what was written before, and probably also more efficient!